armi/reactor/converters/axialExpansionChanger/axialExpansionChanger.py

```diff
@@ -505,71 +489,125 @@ def _addMassToComponent(self, fromComp: "Component", toComp: "Component", deltaZ
-            )
+            try:
+                newToCompTemp = brentq(
+                    f=lambda T: toComp.getArea(Tc=T) - targetArea, a=fromComp.temperatureInC, b=toComp.temperatureInC
+                )
+            except ValueError:
+                totalMass = massFrom + massTo
+                newToCompTemp = (
+                    massFrom / totalMass * fromComp.temperatureInC + massTo / totalMass * toComp.temperatureInC
+                )
+                if (toComp.hasFlags(Flags.FUEL) or toComp.hasFlags(Flags.CONTROL)) or (
+                    fromComp.hasFlags(Flags.FUEL) or fromComp.hasFlags(Flags.CONTROL)
+                ):
+                    msg = f"""
+                    Temperature search algorithm in axial expansion has failed in {self.linked.a}
+                    Trying to search for new temp between
+                        from --> {fromComp.parent} : {fromComp} : {type(fromComp.material)} at {fromComp.temperatureInC} C
+                        to --> {toComp.parent} : {toComp} : {type(toComp.material)} at {toComp.temperatureInC} C
+
+                    f({fromComp.temperatureInC}) = {toComp.getArea(Tc=fromComp.temperatureInC) - targetArea}
+                    f({toComp.temperatureInC}) = {toComp.getArea(Tc=toComp.temperatureInC) - targetArea}
+
+                    Instead, a mass weighted average temperature of {newToCompTemp} will be used. The consequence is that
+                    mass conservation is no longer guaranteed for this component type on this assembly!
+                    """  # noqa: E501
+                    runLog.warning(dedent(msg), label="Temp Search Failure")
+            except Exception as ee:
+                raise ee
+
         # Do not use component.setTemperature as this mucks with the number densities we just calculated.
         toComp.temperatureInC = newToCompTemp
         toComp.clearCache()
 
-    def _removeMassFromComponent(self, fromComp: "Component", deltaZTop: float):
-        """Given ``deltaZTop``, remove mass from ``fromComp``.
+        # update BOL Params for fromComp
+        fromComp.p.molesHmBOL *= 1.0 - (abs(deltaZTop) / fromComp.parent.p.heightBOL)
+        fromComp.p.massHmBOL *= 1.0 - (abs(deltaZTop) / fromComp.parent.p.heightBOL)
 
-        Parameters
-        ----------
-        fromComp
-            Component which is going to give mass to toComp
-        deltaZTop
-            The length, in cm, of fromComp being given to toComp
+    @staticmethod
+    def sortKey(item):
+        match = re.search(r"([a-zA-Z]{1,2})(\d{1,3})([a-zA-Z]?)", item)
+        if match:
+            # Convert numeric parts to int for correct numerical sorting
+            element = match.group(1)
+            atomicWeight = int(match.group(2))
+            metastable = 1 if match.group(3) else 0
+            return (atomicWeight, element, metastable)
+        raise RuntimeError("Unknown isotope!")
```

Comment:
It's useful to tell people why something failed when we can
```suggestion
        raise RuntimeError(f"Unknown isotope - {item}")
```